source/parser.hera

```diff
@@ -7951,33 +7952,102 @@ Init
         let   [ , arg ]   = s.children
 
         let i = 0, l = body.length
-        const children = [ws]
+
+        const refDec = []
+        const children = [ws, refDec]
+
+        let usingRef = null
 
         for (i = 0; i < l; i++) {
           const step = body[i]
           const [leadingComment, pipe, trailingComment, expr] = step
           const returns = pipe.token === "||>"
+          let ref, result,
+            returning = returns ? arg : null
 
           if (pipe.token === "|>=") {
+            let initRef
             if (i === 0) {
-              // TODO: assignment node
-              children.push(arg, " = ")
+              switch(arg.type) {
+                case "MemberExpression":
+                  // If there is only a single access then we don't need a ref
+                  if (arg.children.length <= 2) break
+                case "CallExpression":
+                  const access = arg.children.pop()
```

Comment:
Don't you need to check whether this is an access, not a call? Or because this is a left-hand side it can't be a call?

I think it can also be a "?" or a "!" which you probably want to deal with... There's related code in the object literal shorthand. (I'm glad my work for that to canonicalize the children of a MemberExpression/CallExpression is paying off. But we might want to extract a helper function to find rhe last access...)

Reply:
Yeah, I should check the `access` type. If it is a property or slice then it is fine, but other things not so much.